src/param.rs

```diff
@@ -1088,7 +1088,11 @@ impl AudioParamProcessor {
             match some_event {
                 None => {
                     if is_a_rate {
-                        self.buffer.resize(count, self.intrinsic_value);
+                        let buffer = [self.intrinsic_value; RENDER_QUANTUM_SIZE];
```

Comment:
Maybe it's more succinct to write as:

```rust
for _ in self.buffer.len() .. count {
    self.buffer.try_insert(self.intrinsic_value).unwrap();
}
```

Or did you benchmark this to be the fastest way?

Reply:
Yeah, the L2 and RAM accesses are quite unstable. But since these numbers are so low I don't think it is very important (and that's probably also the reason for the large deviations). I think instruction count is the main metric to track.

I have read https://kobzol.github.io/rust/rustc/2023/09/23/rustc-runtime-benchmarks.html recently with some tips to look further into:

> After the initial refactoring was completed, I needed to decide how will we actually define the benchmarks and what tool we should use to gather the execution metrics. Both cargo bench and [criterion](https://github.com/bheisler/criterion.rs) are not a bad choice for running benchmarks, but they only measure wall-time, while I also wanted to measure hardware counters. I was considering to use [iai](https://github.com/bheisler/iai) for a while. However, it uses [Cachegrind](https://valgrind.org/docs/manual/cg-manual.html) for the measurements, while I wanted the benchmarks to be executed natively, without simulation. Also, using Cachegrind wouldn’t produce realistic wall-time results.

> In the end, I decided to write a small library called [benchlib](https://github.com/rust-lang/rustc-perf/tree/1d1400b66e232cd968c3598cb725c4133cea63a3/collector/benchlib), so that we would have ultimate control of defining, executing and measuring the benchmarks, instead of relying on external crates. benchlib uses Linux perf events to gather hardware metrics, using the [perf-event](https://github.com/jimblandy/perf-event) crate. I also took bits and pieces from other mentioned tools, like the [black_box](https://github.com/rust-lang/rustc-perf/blob/414230abc695bd715054d633c8203090f7423ce5/collector/benchlib/src/benchmark.rs#L161) function from iai.